cylc/flow/commands.py

```diff
@@ -314,11 +323,15 @@ async def set_verbosity(schd: 'Scheduler', level: Union[int, str]):
 
 
 @_command('remove_tasks')
-async def remove_tasks(schd: 'Scheduler', tasks: Iterable[str]):
+async def remove_tasks(
+    schd: 'Scheduler', tasks: Iterable[str], flow: List[str]
+):
     """Remove tasks."""
     validate.is_tasks(tasks)
+    validate.flow_opts(flow, flow_wait=False, allow_new_or_none=False)
     yield
-    yield schd.pool.remove_tasks(tasks)
+    flow_nums = get_flow_nums_set(flow)
```

Comment:
Would it make sense to put this before the yield so any failure happens in the validate step?

Reply:
I wouldn't try moving that line.

E.G, Imagine there are two queued commands:
* trigger(foo, flow=new)
* trigger(bar, flow=all)

At present we would get:
* foo (flow=2)
* bar (flow=1,2)

If we moved this up a line, we would get:
* foo (flow=2)
* bar (flow=1)